diagnostics_tools/src/periodic_event_status.cpp

```diff
@@ -4,12 +4,14 @@
 
 namespace qna {
 namespace diagnostic_tools {
-
 PeriodicEventStatus::PeriodicEventStatus(const std::string& name)
-    : PeriodicEventStatus(name, PeriodicEventStatusParams{}) {}
+    : PeriodicEventStatus(name, PeriodicEventStatusParams{}){}
 
 PeriodicEventStatus::PeriodicEventStatus(const std::string& name, PeriodicEventStatusParams params)
-    : PeriodicDiagnosticTask(name), params_(std::move(params)) {}
+    : PeriodicDiagnosticTask(name),
+      params_(std::move(params)),
+      last_cycle_period_(params.last_cycle_period_avg_window()),
+      historic_period_(params.historic_period_avg_window()){}
```

Comment:
@patapufate as far as I can see, this diagnostic check still resets windows on every tick.

Reply:
Only reset if "Time went backwards"

Reply:
Although I removed from run method.